applications/ContactStructuralMechanicsApplication/custom_processes/mpc_contact_search_wrapper_process.cpp

```diff
@@ -0,0 +1,112 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Vicente Mataix
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+/* Processes */
+#include "custom_processes/mpc_contact_search_process.h"
+#include "custom_processes/mpc_contact_search_wrapper_process.h"
+
+namespace Kratos
+{
+MPCContactSearchWrapperProcess::MPCContactSearchWrapperProcess(
+    ModelPart& rMainModelPart,
+    Parameters ThisParameters
+    )
+{
+    // The default parameters
+    Parameters default_parameters = GetDefaultParameters();
+    ThisParameters.ValidateAndAssignDefaults(default_parameters);
+
+    // The dimensions
+    const SizeType dimension = rMainModelPart.GetProcessInfo()[DOMAIN_SIZE];
+    const bool predefined_master_slave = ThisParameters["predefined_master_slave"].GetBool();
+    SizeType size_1 = 0;
+    SizeType size_2 = 0;
+    for (auto& r_cond : rMainModelPart.Conditions()) {
+
+        if (predefined_master_slave) {
+            if (r_cond.Is(MASTER))
+                size_1 = r_cond.GetGeometry().size();
+            if (r_cond.Is(SLAVE))
+                size_2 = r_cond.GetGeometry().size();
+        } else {
+            size_1 = r_cond.GetGeometry().size();
+            size_2 = size_1;
+        }
+
+        // Once both defined we break
+        if (size_1 > 0 && size_2 > 0)
+            break;
+    }
+
+    // Removing to avoid problems
+    ThisParameters.RemoveValue("simple_search");
```

Comment:
why don't you just ignore it... it is generally not nice to remove entries from settings (i am aware it is done elsewhere...still it is not nice)

Reply:
This is done in the other wrapper, I can remove it in both places in a future PR